### PR TITLE
Error Prone: Fix catch and print stack trace violations

### PR DESCRIPTION
**Pull request by ssoloff**
## Overview

Fixes violations of the Error Prone CatchAndPrintStackTrace rule.

## Functional Changes

See forthcoming review.

## Manual Testing Performed

None.

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/ui/AbstractUiContext.java [hunk 1]
```diff
@@ -283,7 +283,7 @@ public void setLockMap(final boolean lockMap) {
     try {
       prefs.flush();
     } catch (final BackingStoreException ex) {
-      ex.printStackTrace();
+      ClientLogger.logQuietly("Failed to flush preferences: " + prefs.absolutePath(), ex);
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/triplea/ui/AbstractUiContext.java]: Previously we logged to stderr, and now we log to stdout.  If you look at similar exception handlers in this file, you'll see this change is consistent with the others.

File: game-core/src/main/java/games/strategy/triplea/ui/AbstractUiContext.java [hunk 2]
```diff
@@ -317,7 +317,7 @@ public void setShowTriggeredNotifications(final boolean value) {
...
```
Comment (ssoloff) [inline game-core/src/main/java/games/strategy/triplea/ui/AbstractUiContext.java]: Ditto.

File: game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java [hunk 3]
```diff
@@ -275,8 +275,8 @@ private void warmUpCryptoRandomSource() {
     new Thread(() -> {
       try {
         serverGame.getRandomSource().getRandom(gameData.getDiceSides(), 2, "Warming up crypto random source");
-      } catch (final RuntimeException re) {
-        re.printStackTrace(System.out);
+      } catch (final RuntimeException e) {
+        ClientLogger.logQuietly("Failed to warm up crypto random source", e);
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java]: I'm not sure, is this shared code between headless and non-headless? (or non-headless only?)

Comment (ssoloff) [reply game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java]: I believe this class is used by both the headed and headless game servers.  Other methods in this class are already using `ClientLogger` (not claiming that's correct, but it's precedent).

**Pull request merged**